core/unit_test/cuda/TestCuda_TeamScratch.cpp

```diff
@@ -47,6 +47,91 @@
 
 namespace Test {
 
+namespace Impl {
+
+void cuda_stream_scratch_test_one(
+    int N, int T, int M_base, Kokkos::View<int64_t, Kokkos::CudaSpace> counter,
```

Comment:
I would have made it an atomic view to do just `++counter()` below.  You don't have to change.